src/index.ts

```diff
@@ -1,130 +1,234 @@
     {
-      title: 'Encoding patched APK file',
-      task: () => new Listr([
-        {
-          title: 'Encoding using AAPT2',
-          task: (_, task) => new Observable(subscriber => {
-            apktool.encode(decodeDir, unsignedApkPath, true).subscribe(
-              line => subscriber.next(line),
-              () => {
-                subscriber.complete()
-                task.skip('Failed, falling back to AAPT...')
-                fallBackToAapt = true
-              },
-              () => subscriber.complete(),
-            )
-          }),
-        },
-        {
-          title: chalk`Encoding using AAPT {dim [fallback]}`,
-          skip: () => !fallBackToAapt,
-          task: () => apktool.encode(decodeDir, unsignedApkPath, false),
-        }
-      ]),
+      title: "Encoding patched APK file",
+      task: () =>
+        new Listr([
+          {
+            title: "Encoding using AAPT2",
+            task: (_, task) =>
+              new Observable(subscriber => {
+                apktool.encode(decodeDir, unsignedApkPath, true).subscribe(
+                  line => subscriber.next(line),
+                  () => {
+                    subscriber.complete();
+                    task.skip("Failed, falling back to AAPT...");
+                    fallBackToAapt = true;
+                  },
+                  () => subscriber.complete()
+                );
+              })
+          },
+          {
+            title: chalk`Encoding using AAPT {dim [fallback]}`,
+            skip: () => !fallBackToAapt,
+            task: () => apktool.encode(decodeDir, unsignedApkPath, false)
+          }
+        ])
     },
     {
-      title: 'Signing patched APK file',
-      task: () => new Observable(subscriber => {
-        (async () => {
-          await uberApkSigner.sign(unsignedApkPath)
-            .forEach(line => subscriber.next(line))
-
-          await fs.copyFile(
-            path.join(tmpDir, 'unsigned-aligned-debugSigned.apk'),
-            finishedApkPath,
-          )
-
-          subscriber.complete()
-        })()
-      }),
-    },
-  ]).run().catch(error => {
-    console.error(
-      chalk`\n  {red.inverse.bold  Failed! } An error occurred:\n\n`,
-      error.toString(),
-    )
-
-    process.exit(1)
-  })
+      title: "Signing patched APK file",
+      task: () =>
+        new Observable(subscriber => {
+          (async () => {
+            await uberApkSigner
+              .sign(unsignedApkPath)
+              .forEach(line => subscriber.next(line));
+
+            await fs.copyFile(
+              path.join(tmpDir, "unsigned-aligned-debugSigned.apk"),
+              finishedApkPath
+            );
+
+            subscriber.complete();
+          })();
+        })
+    }
+  ])
+    .run()
+    .catch(error => {
+      console.error(
+        chalk`\n  {red.inverse.bold  Failed! } An error occurred:\n\n`,
+        error.toString()
+      );
+
+      process.exit(1);
+    });
 
   if (usesAppBundle) {
```

Comment:
Checking this variable and trying App Bundle modification if it is `true` is not going to work as the input files are entirely different (single APK file vs. zip file containing multiple APK files). Instead, we should tell the user to run `apk-mitm` again with a `.xapk` or `.apks` file.

As a result, having both a `prepareApk` and `prepareAppBundle` function (the latter using `prepareApk` internally) would make sense. Inside of `cli.js` we could then call the appropriate one based on the input file's extension (`.apk` for `prepareApk` and `.xapk`, `.apks`, and `.zip` for `prepareAppBundle`).

Reply:
Done ✅